pkg/ttnpb/lorawan.go

```diff
@@ -36,55 +35,93 @@ func init() {
 	PHYVersion_value["1.1-b"] = int32(PHY_V1_1_REV_B)   // 1.1 is the official version number
 }
 
+// MarshalBinary implements encoding.BinaryMarshaler interface.
+func (v MType) MarshalBinary() ([]byte, error) {
+	if v > 255 {
+		panic(fmt.Errorf("MType enum exceeds 255"))
```

Comment:
Why panic when you can also return an error?

Reply:
more below

Reply:
This panic indicates an implementation error, not input error, the enum is expected to be valid - i.e. be way below 255.

Reply:
You don't know that in `pkg/ttnpb`. We import this package in other projects, like the AWS IoT Integration, and panics because of marshaling are not expected there, especially when there's an error result available.

It could be true for NS, but then you can wrap it in a `MustMarshalBinary()` if you really feel you don't want to add the `if err != nil { return [nil, ]err }` line all the time